library/test/src/test/java/com/bumptech/glide/load/resource/gif/GifDrawableTest.java

```diff
@@ -568,12 +569,53 @@ public void testSetAlphaSetsAlphaOnPaint() {
   public void testSetColorFilterSetsColorFilterOnPaint() {
     ColorFilter colorFilter = new PorterDuffColorFilter(Color.RED, Mode.ADD);
     drawable.setColorFilter(colorFilter);
+    verify(paint).setColorFilter(eq(colorFilter));
+  }
+
+  @Config(sdk = Build.VERSION_CODES.LOLLIPOP)
+  @Test
+  public void testDrawSetsTintListColorFilterOnPaint() {
+    ColorStateList tint =
+        new ColorStateList(
+            new int[][] {new int[] {android.R.attr.state_pressed}, new int[0]},
+            new int[] {Color.RED, Color.GREEN});
+    drawable.setTintList(tint);
+    drawable.setTintMode(Mode.ADD);
+    when(paint.getColorFilter()).thenReturn(null);
+    drawable.draw(new Canvas());
+
+    // draw() temporary sets tint filter then restore.
+    verify(paint).setColorFilter(eq(new PorterDuffColorFilter(Color.GREEN, Mode.ADD)));
+    verify(paint).setColorFilter(null);
```

Comment:
this test is not testing drawable behavior but internal implementation details of the draw method, what we care about is that the bitmap was drawn with a paint configured with the tint filter, not that paint was interacted with.

Reply:
Thank  you for pointing it out!
Previously I followed the test code of setColorFilter().

Now, I updated the test code of both setColorFilter() and setTintList() to check the Paint object which is used by Canvas#drawBitmap().
PTAL?

(Ideally it's better to check the drawn bitmap colors, but we cannot do it on Robolectric tests.)

Reply:
Note:
I updated the behavior of GifDrawable#draw() to make it test-friendly.

We had the following code inside GifDrawable#draw().

>  // Temporary set a tint filter then restore.
> paint.setColorFilter(tintFilter);
> canvas.drawBitmap(currentFrame, null, getDestRect(), paint);
> paint.setColorFilter(colorFilter);

With this code, we cannot verify whether `canvas#drawBitmap()` uses the `tintFilter` or not with ArgumentCaptor, because `paint.setColorFilter(colorFilter)` is called before `ArgumentCaptor#capture()`. The new code doesn't call `paint.setColorFilter(colorFilter)` to avoid the issue.